### DISCUSSION TITLE
how do you identify the geometry type in the pmtiles file?

### DISCUSSION THREAD

**Discussion by drzhouq** [protomaps/PMTiles#159] [Q&A] [answered]
It seems the metadata doesn't have that information. Without that, how can one specify the painter_rules?
Thanks for the help

**Comment by bdon** [comment, marked as answer]
by convention you can include `vector_layers` in the JSON metadata, this will be part of spec revision 3.1: https://github.com/protomaps/PMTiles/pull/123

That will give you the necessary information to write `paint_rules` or MapLibre JSON styles. But the MVT spec lets any type of geometry be in each layer.

**Comment by bdon** [comment]
(convert to discussion)
